inject/android.js

```diff
@@ -35,13 +35,14 @@ function initCode () {
         if (providedSecret === messageSecret) {
             update(...args)
         }
+        // eslint-disable-next-line no-extra-bind
     }).bind(this)
 
     Object.defineProperty(window, messageCallback, {
         value: wrappedUpdate
     })
 
-    // @ts-expect-error
+    // @ts-expect-error - Android message interface defined elsewhere
     const sendMessageToAndroid = window[messageInterface].process.bind(window[messageInterface])
```

Comment:
Was this the issue why it was broken?

Whatever it was can we account for it not being present if we wish to land it in isolation?

Reply:
I've been looking at this too, I think it's better to remove the global interface entirely, like this: 

```js
const originalHandler = window[messageInterface]
const sendMessageToAndroid = originalHandler.process.bind(originalHandler)
Reflect.deleteProperty(window, messageInterface) // doesn't throw

```

Reply:
I don't understand @shakyShane , are you saying `Reflect.deleteProperty(window, messageInterface)` is better than `delete window[messageInterface]` here?

Reply:
Ahh sorry @kzar I mis-read - I was going to make the point that you should remove the entire thing, not just process, but I can see you are doing that 👍🏻

regardless I think you need the reference to window.<interface> for the binding to live on after it's deleted, that's why I did the binding in 2 steps as shown above.

Might be wrong on that, I can verify :)

Reply:
Updated my comment, just verified this approach on Android and it works :) sorry for the noise, I thought I'd encountered issues doing it this way in the past, but this looks fine in the simulator :)

Reply:
Can we answer these:

- Was this the issue why it was broken (and was reverted)?
- Whatever it was can we account for it not being present if we wish to land it in isolation (to the Android changes)?

Reply:
@jonathanKingston 
- Yes, it was looking for the interface which was not merged yet on the Android side
- I've since merged the Android messaging changes which means the interface is now there: https://github.com/duckduckgo/Android/pull/3083
- And I've run the action to check if this branch will cause any problems with the privacy tests: https://github.com/duckduckgo/Android/actions/runs/4758472939/jobs/8456577141
- So merging this is safe ✅

Reply:
I think @jonathanKingston  has a point though, it wouldn't harm to first check `window[messageInterface]` exists before attempting to use it. How about something like this?

```javascript
let sendMessageToAndroid
if (Object.prototype.hasOwnProperty.call(window, messageInterface)) {
  sendMessageToAndroid = window[messageInterface].process.bind(window[messageInterface])
  delete window[messageInterface]
} else {
  sendMessageToAndroid = () => { console.error('Android messaging interface not available') }
}
```

Reply:
I added that and it doesn't break the rest of the script now if the interface is not present 👍
I still needed to suppress `process` not existing on the window, so I made the explanation more explicit.